src/xGadgetron/cGadgetron/tests/CMakeLists.txt

```diff
@@ -17,13 +17,53 @@
 #
 #=========================================================================
 
+
+function(get_zenodo_file path url md5)
+        message(STATUS "Downloading (if not already done)...
+                Path:     ${path}
+                URL:      ${url}
+                MD5:      ${md5}")
+        file(DOWNLOAD "${url}" "${path}" SHOW_PROGRESS EXPECTED_MD5 "${md5}")
+endfunction()
+
+OPTION(DOWNLOAD_ZENODO_TEST_DATA "download zenodo data for tests" ON)
+if (DOWNLOAD_ZENODO_TEST_DATA)
+        set(MR_RPE_DATA "${CMAKE_SOURCE_DIR}/data/examples/MR/zenodo/3D_RPE_Lowres.h5")
+        get_zenodo_file(
+                "${MR_RPE_DATA}"
+                "https://zenodo.org/record/4600937/files/3DRPE_Lowres.h5?download=1"
+                "a692ba3a6c456d1ad49750717a9d5dda")
+endif()
+
+OPTION(RERUN_ISMRMRD_SHEPP_LOGAN "Generate testdata compatible with ISMRMRD Version" ON)
+find_program(EXE_ISMRMRD_SL NAMES ismrmrd_generate_cartesian_shepp_logan
+                NAMES_PER_DIR
+                PATHS ${CMAKE_INSTALL_PREFIX})
+
+if(EXE_ISMRMRD_SL AND RERUN_ISMRMRD_SHEPP_LOGAN)
+    add_custom_target(GENERATE_SIMULATED_TESTDATA ALL
+        COMMAND rm -f simulated_MR_2D_cartesian.h5
+        COMMAND rm -f simulated_MR_2D_cartesian_Grappa2.h5
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -r 2 -o simulated_MR_2D_cartesian.h5
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -a 2 -o simulated_MR_2D_cartesian_Grappa2.h5
+        WORKING_DIRECTORY "${SIRF_SOURCE_DIR}/data/examples/MR/")
```

Comment:
@evgueni-ovtchinnikov , can we add your "fixing-up" code here?

Reply:
@KrisThielemans: if you mean the one that computes non-zero slice directions then yes, we just need to send the outputs of line 47 and 48 to some temporary files, e.g. `tmp1.h5` and `tmp2.h5`, and make CMake run `MR_TESTS_CPLUSPLUS` binary first with arguments `tmp1.h5` and `simulated_MR_2D_cartesian.h5` and then with `tmp2.h5` and `simulated_MR_2D_cartesian_Grappa2.h5`, and delete `tmp1.h5` and `tmp2.h5`. @johannesmayer: can you do that, please (and also remove printing from `edit_acq`)?

Reply:
sure, I am currently in the CMakeCrash course anyway so when this is finished I will be an expert on that matter!

Reply:
Alright, I added an executable and run it to preprocess the data and made the MR C++ tests depend on that target.
So far, this preprocessing only sets the acquisition orientation, but if in the future something comes up we could add it to that preprocessing.